File: i2f-springboot/i2f-springboot-swl-starter/src/main/java/i2f/springboot/swl/filter/SwlWebConfig.java
package i2f.springboot.swl.filter;

import i2f.swl.impl.SwlBase64Obfuscator;
import i2f.swl.impl.SwlSha256MessageDigester;
import i2f.swl.impl.supplier.SwlAesSymmetricEncryptorSupplier;
import i2f.swl.impl.supplier.SwlRsaAsymmetricEncryptorSupplier;
import i2f.swl.std.ISwlMessageDigester;
import i2f.swl.std.ISwlObfuscator;
import i2f.swl.std.supplier.ISwlAsymmetricEncryptorSupplier;
import i2f.swl.std.supplier.ISwlSymmetricEncryptorSupplier;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * @author Ice2Faith
 * @date 2024/7/10 15:36
 * @desc
 */
@Data
@NoArgsConstructor
public class SwlWebConfig {
    protected SwlWebCtrl defaultCtrl = new SwlWebCtrl(true, true);
    protected String headerName = "swlh";
    protected String remoteAsymSignHeaderName = "swlras";
    protected String currentAsymKeyHeaderName = "swlcak";
    protected String realContentTypeHeaderName = "swlct";
    protected String parameterName = "swlp";
    protected String responseCharset = "UTF-8";
    protected boolean filterResponseException=false;
    protected List<String> whiteListIn;
    protected List<String> whiteListOut;
    protected List<String> attachedHeaderNames;
    protected Class<? extends ISwlAsymmetricEncryptorSupplier> asymAlgoClass = SwlRsaAsymmetricEncryptorSupplier.class;
    protected Class<? extends ISwlSymmetricEncryptorSupplier> symmAlgoClass = SwlAesSymmetricEncryptorSupplier.class;
    protected Class<? extends ISwlMessageDigester> digestAlgoClass = SwlSha256MessageDigester.class;
    protected Class<? extends ISwlObfuscator> obfuscateAlgoClass= SwlBase64Obfuscator.class;
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/filter/SwlWebConfig.js
import SwlWebCtrl from "./SwlWebCtrl";
import SwlRsaAsymmetricEncryptorSupplier from "../core/impl/supplier/SwlRsaAsymmetricEncryptorSupplier";
import SwlSha256MessageDigesterSupplier from "../core/impl/supplier/SwlSha256MessageDigesterSupplier";
import SwlBase64Obfuscator from "../core/impl/SwlBase64Obfuscator";
import SwlAesSymmetricEncryptorSupplier from "../core/impl/supplier/SwlAesSymmetricEncryptorSupplier";

/**
 * @return SwlWebConfig
 * @constructor
 */
function SwlWebConfig() {
    /**
     * @type {boolean}
     */
    this.enable=true;
    /**
     * @type {SwlWebCtrl}
     */
    this.defaultCtrl = new SwlWebCtrl(true, true);
    /**
     * @type {String}
     */
    this.headerName = "swlh";
    /**
     * @type {String}
     */
    this.certIdName = "swlci";
    /**
     * @type {String}
     */
    this.realContentTypeHeaderName = "swlct";
    /**
     * @type {String}
     */
    this.parameterName = "swlp";
    /**
     * @type {String}
     */
    this.responseCharset = "UTF-8";
    /**
     * @type {boolean}
     */
    this.filterResponseException = false;
    /**
     *
     * @type {string[]}
     */
    this.urlPatterns = null;
    /**
     *
     * @type {string[]}
     */
    this.whiteListIn = ['/**/swl/**'];
    /**
     *
     * @type {string[]}
     */
    this.whiteListOut = ['/**/swl/**'];

    /**
     *
     * @type {string[]}
     */
    this.attachedHeaderNames = null;
    /**
     * @type {ISwlAsymmetricEncryptorSupplier}
     */
    this.asymAlgoSupplier = new SwlRsaAsymmetricEncryptorSupplier();
    /**
     *
     * @type {ISwlSymmetricEncryptorSupplier}
     */
    this.symmAlgoSupplier = new SwlAesSymmetricEncryptorSupplier();
    /**
     *
     * @type {ISwlMessageDigesterSupplier}
     */
    this.digestAlgoSupplier = new SwlSha256MessageDigesterSupplier();
    /**
     *
     * @type {ISwlObfuscator}
     */
    this.obfuscateAlgoSupplier = new SwlBase64Obfuscator();
}

export default SwlWebConfig
